### PR TITLE
Feat/90/new request approval/rejection

### PR DESCRIPTION
**Pull request by mrstrval**

### CODE REVIEW

File: server/db-middleware/test-api.js [hunk 1]
```diff
@@ -3,35 +3,35 @@ const {
...
+    .then((count) => {
+      if (!count) {
+        return Section.findById(sectId);
+      }
+    });
 };
 
-apiModule.acceptableSectionsToRequest = function (userId) {
-  let result = [];
+apiModule.acceptableSectionsToRequest = function (studentId) {
```
Comment (whybazinga) [inline server/db-middleware/test-api.js]: getAcceptableSectionsToRequest

File: server/db-middleware/test-api.js [hunk 2]
```diff
@@ -3,35 +3,35 @@ const {
...
-    .then((sections) => {
-      result = sections;
-      return apiModule.getStudentRequestsWithStatus(userId, [REQUEST_STATUS_APPROVED, REQUEST_STATUS_PENDING]);
-    })
-    .then(allUnacceptableRequests => Promise.all(allUnacceptableRequests.map(el => this.getSectionByRequestId(el))))
-    .then(checking => result.filter(object => checking.every((element) => {
-      if (element.id === object.id) { return false; }
+    .then(sections => Promise.all(sections.map(el => checkRequestsForSections(el.id, studentId))))
+    .then(sections => sections.filter((object) => {
+      if (object == null) return false;
```
Comment (whybazinga) [inline server/db-middleware/test-api.js]: Be careful using ==, it has pretty unobvious behavior.
Also do you really need to compare with null? I suppose if (object) will aslo work.

File: server/db-middleware/test-api.js [hunk 3]
```diff
@@ -42,4 +42,45 @@ apiModule.newTestRequest = function (user, section) {
   });
 };
 
+apiModule.rejectRequest = function (requestId) {
+  return Request
+    .findById(requestId, (err, request) => {
```
Comment (whybazinga) [inline server/db-middleware/test-api.js]: Check findByIdAndUpdate method in mongoose documentation, consider using it.

File: server/db-middleware/test-api.js [hunk 4]
```diff
@@ -42,4 +42,45 @@ apiModule.newTestRequest = function (user, section) {
...
+  return TagAttachment
+    .find({ sectionId: sectId })
+    .then(tags => Promise.all(tags.map(el => el.tag)));
+};
+
+apiModule.approveRequest = function (requestId, teachId) {
+  let requestToRemember;
+  let sectionName;
+  return Request
+    .findById(requestId, (err, request) => {
```
Comment (whybazinga) [inline server/db-middleware/test-api.js]: findByIdAndUpdate

File: server/models/testing/test-assignment.js [hunk 5]
```diff
@@ -7,6 +7,7 @@ const testAssignmentSchema = new Schema({
   studentId: { type: Schema.Types.ObjectId, ref: 'User', index: true },
   name: { type: String, required: true },
   tags: { type: [{ type: String }], index: true, required: true },
+  sectionId: { type: Schema.Types.ObjectId, index: true, ref: 'Section' },
```
Comment (whybazinga) [inline server/models/testing/test-assignment.js]: There should be no sectionId. In this case you should fill Test-assignment's tags field with your section's tags.

File: server/db-middleware/test-api.js (hunk 2 above)
Comment (mrstrval) [reply server/db-middleware/test-api.js]: Boolean (object) is always true.
Also it should be there to specify, that variable was not defined.

File: server/db-middleware/test-api.js [hunk 6]
```diff
@@ -29,7 +29,10 @@ apiModule.getAcceptableSectionsToRequest = function (studentId) {
     .find()
     .then(sections => Promise.all(sections.map(el => checkRequestsForSections(el.id, studentId))))
     .then(sections => sections.filter((object) => {
-      if (object == null) return false;
+      if (!object) {
+        console.log(object);
```
Comment (whybazinga) [inline server/db-middleware/test-api.js]: Delete console.log

**Pull request merged**